app/src/main/java/com/sethpthomas91/httpserver/handlers/HeadRequestHandler.java

```diff
@@ -0,0 +1,58 @@
+package com.sethpthomas91.httpserver.handlers;
+
+import com.sethpthomas91.httpserver.interfaces.HttpRequestInterface;
+import com.sethpthomas91.httpserver.response.Body;
+import com.sethpthomas91.httpserver.response.Header;
+import com.sethpthomas91.httpserver.response.HttpResponseWrapper;
+import com.sethpthomas91.httpserver.response.StatusLine;
+
+import java.io.IOException;
+
+public class HeadRequestHandler implements Handler {
+    private String[] allowedMethods = {"HEAD", "OPTIONS"};
+
+    @Override
+    public HttpResponseWrapper handle(HttpRequestInterface httpRequest) throws IOException {
+        HttpResponseWrapper httpResponse = new HttpResponseWrapper();
+        httpResponse = handleStatusLine(httpRequest, httpResponse);
+        httpResponse = handleBody(httpRequest, httpResponse);
+        httpResponse = handleHeaders(httpRequest, httpResponse);
+        return httpResponse;
+    }
+
+    private HttpResponseWrapper handleStatusLine(HttpRequestInterface httpRequest, HttpResponseWrapper httpResponse) {
+        StatusLine statusLine = new StatusLine();
+        switch (httpRequest.getRequestLine().getTypeOfRequest()) {
+            case "GET":
+                statusLine.setStatusCode("405");
+                statusLine.setHttpVersion(httpRequest.getRequestLine().getHttpVersion());
+                statusLine.setResponseText("OK");
+                break;
+            case "HEAD":
+                statusLine.setStatusCode("200");
+                statusLine.setHttpVersion(httpRequest.getRequestLine().getHttpVersion());
+                statusLine.setResponseText("OK");
+                break;
+        }
+        httpResponse.setStatusLine(statusLine);
+        return httpResponse;
+    }
+
+    private HttpResponseWrapper handleBody(HttpRequestInterface httpRequest, HttpResponseWrapper httpResponse) throws IOException {
+        Body body = new Body();
+        httpResponse.setBody(body);
+        return httpResponse;
+    }
+
+    private HttpResponseWrapper handleHeaders(HttpRequestInterface httpRequest, HttpResponseWrapper httpResponse) throws IOException {
+        Header header = new Header(httpRequest, httpResponse);
+        header.setAllowedHeaders("HEAD, OPTIONS");
```

Comment:
These could be changed to be dynamically created from the private variable to be one source of truth. This could be applied to the other handlers as well.